python/cog/server/runner.py

```diff
@@ -276,6 +281,37 @@ def _upload_files(self, output: Any) -> Any:
             raise FileUploadError("Got error trying to upload output files") from error
 
 
+def setup(*, worker: Worker):
+    logs = []
+    status = None
+    started_at = datetime.now(tz=timezone.utc)
+
+    try:
+        for event in worker.setup():
```

Comment:
This still has to finish before it shuts down, right? Or do the signals/events propegate in such a way that we interrupt the model booting, before we shut it down? Not obvious to me

Reply:
This is running in a thread managed by `PredictionRunner`. When director requests a shutdown, `PredictionRunner.shutdown()` is called, which terminates the worker process and the thread running this code: https://github.com/replicate/cog/blob/d574a3496947eea7ae09929c2544c80bbf23e7d7/python/cog/server/runner.py#LL138C7-L138C7

So no, it should shut straight down, and testing that locally seems to validate that.